lib/kubernetes-deploy/runner.rb

```diff
@@ -283,12 +279,15 @@ def validate_namespace
       KubernetesDeploy.logger.info("Namespace #{@namespace} validated")
     end
 
-    def run_kubectl(*args, namespaced: true)
+    def run_kubectl(*args, namespaced: true, with_context: true)
       args = args.unshift("kubectl")
       if namespaced
-        raise FatalDeploymentError, "Namespace missing for namespaced command" unless @namespace
```

Comment:
💯  on the `--context` flag, but why did you remove this safeguard? It's true that we shouldn't ever hit it because of up-front validations, but it exists because leaving out the namespace would be very, very bad. Same thing for context, really.